src/annotator/integrations/vitalsource.js

```diff
@@ -110,10 +82,25 @@ export class VitalSourceInjector {
         return;
       }
       contentFrames.add(frame);
-
-      injectIfContentReady(frame);
-      frame.addEventListener('load', () => {
-        injectIfContentReady(frame);
+      onDocumentReady(frame, (err, document_) => {
```

Comment:
nb. We don't check `err` here, instead we rely on `document_` being undefined if there was an error. In that case `isBookContent` will be false.